Sources/GRPC/AsyncAwaitSupport/GRPCAsyncServerHandler.swift

```diff
@@ -0,0 +1,519 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import _NIOConcurrency
+import Logging
+import NIOCore
+import NIOHPACK
+
+#if compiler(>=5.5)
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public final class GRPCAsyncServerHandler<
```

Comment:
OK, I must have misremembered or conflated this with another discussion because I did this (excluding the protocol) deliberately based on my (potentially flawed) memory of what was asked.

Reply:
I'm not sure if we actually discussed this to be honest 🤷‍♂️

Reply:
Either way, I'll rework this to conform to a protocol 👍.

Reply:
I've had a crack at this in b73edc3e and c80b35f8. LMK if this was what you meant.

Reply:
Looks good!